### PR TITLE
Add Dockerfile for Syson

### PR DESCRIPTION
**Pull request by GorgiAstro**
This merge request introduces a new Dockerfile. Unlike the existing Dockerfile in `backend/application/syson-application`, this Dockerfile builds Syson and its dependencies.

inspired from https://github.com/eclipse-syson/syson/discussions/47

The files should maybe be placed somewhere else.

### CODE REVIEW

**Comment by AxelRICHARD** [comment]
Hello,
Thank you for your contribution. For your information, the script and video from #47 are not official.
Indeed, I would like to put it in a dedicated root folder (e.g. `docker`).
Some others things that should be done:
- could you open a ticket please so I can associate this PR to the ticket?
- could you change your commit message to indicate the ticket please? (see git history, It will show you how commit messages should be written)
- could you update the changelog please?
- could you separate the build part from the run part please ? With the current version, some dependencies only needed for the build are contained in the run image (e.g. maven). I think you should follow be able to create a multi-stage image to segment the following 3 steps:
1. Build the frontend: From the official Node image.
2. Build the backend: From the Temurin 17 JDK image. Copy the result of the frontend build from the intermediate frontend build image.
4. Run image: From the Temurin 17 JRE image.
Copy the result of the backend build from the intermediate backend build image.
Avoid, as much as possible, the installation of packages from the internet (installation of nvm not mandatory). Use external packages only from Maven or npm install.

Thank you!
Regards,

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
+FROM eclipse-temurin:17-jdk-jammy
+
+ENV NODE_VERSION="18.7"
+ENV SYSON_VERSION="2024.1.0"
```
Comment (jamilraichouni) [inline Dockerfile:4]: I tried to build the container and the value of this env var is wrong. I'd say it must be `2024.3.0`

Comment (AxelRICHARD) [reply Dockerfile:4]: Indeed, this PR has been provided before the release of 2024.3.0.
The author never take into account my comments since then.

Comment (jamilraichouni) [reply Dockerfile:4]: Hi @AxelRICHARD , when I manually fix this in the running container i still get a startup error (see below).
Can you help and tell me what I or the Dockerfile misses here?

Where do I see your comments?

image

**Comment by GorgiAstro** [comment]
Hi Alex, 

Thank you for the comments. Sorry for not answering, these past few weeks have been very busy and your answer has slipped under my radar.
I'll try to rework the docker image as you suggested but I don't know when I'll manage that.

Cheers
Clément

**Comment by jamilraichouni** [comment]
> Hi Alex,
> 
> Thank you for the comments. Sorry for not answering, these past few weeks have been very busy and your answer has slipped under my radar. I'll try to rework the docker image as you suggested but I don't know when I'll manage that.
> 
> Cheers Clément

Hi @GorgiAstro , can you maybe tell me what I respetively the ENTRYPOINT misses (see my comment above).
When I fix the env var and rebuild the Docker container the run fails with that error message.
Thanks!

**Comment by jamilraichouni** [comment]
I corrected the env var value and as reported the Docker container run fails.
It will be highly appreciated to get a tip where I need to look to understand what is missing here. The README is not describing how one starts SysON.

image

**Comment by AxelRICHARD** [comment]
> I corrected the env var value and as reported the Docker container run fails. It will be highly appreciated to get a tip where I need to look to understand what is missing here. The README is not describing how one starts SysON.

You can try the following instructions from the discussions:
https://github.com/eclipse-syson/syson/discussions/83#discussioncomment-8663531

**Comment by GorgiAstro** [comment]
@jamilraichouni did you start the postgresql container before starting Syson?

I just tested building and running the docker image based on the latest syson main branch, and except having to change ENV SYSON_VERSION to "2024.3.0", it worked for me.

**Comment by GorgiAstro** [comment]
Oh I found the issue, like Axel said, you have to pass the SPRING_DATASOURCE_URL among other parameters:

- if using docker compose, for instance:
```
syson:
    image: syson
    container_name: syson
    restart: unless-stopped
    ports:
      - '8080:8080'
    environment:
      SPRING_DATASOURCE_URL: jdbc:postgresql://sysondb/postgres
      SPRING_DATASOURCE_USERNAME: test_username
      SPRING_DATASOURCE_PASSWORD: test_password
      SIRIUS_COMPONENTS_CORS_ALLOWEDORIGINPATTERNS: '*'
      SERVER_PORT: 8080
    depends_on:
      - sysondb
```

and if using `docker run`, refer to the link provided by Alex.

**Comment by GorgiAstro** [comment]
Awesome, thanks a lot!

**Pull request closed without merging**